osbuild/pipeline.py

```diff
@@ -473,10 +495,20 @@ def depsolve(self, store: ObjectStore, targets: Iterable[str]) -> List[str]:
 
         return list(map(lambda x: x.name, reversed(build.values())))
 
-    def build(self, store, pipelines, monitor, libdir, debug_break="", stage_timeout=None):
+    def build(self, store, pipelines, monitor, libdir, debug_break="", stage_timeout=None) -> Dict[str, Any]:
+        """Build the manifest
+
+        Returns a dict of string keys that contains the overall
+        "success" and the `BuildResult` of each individual pipeline.
+
+        The overall success "success" is stored as the string "success"
+        with the bool result and the build pipelines BuildStatus is
+        stored under the pipelines ID string.
+        """
         results = {"success": True}
 
         for pl in map(self.get, pipelines):
+            assert pl is not None
```

Comment:
Would it be better to instead do this in `get()` and error there with some meaningful error type instead of asserting here and hiding where it went wrong?

Reply:
Thanks, this was indeed a bit too terse. It was prompted by mypy but I added a commit now to cleanup this code a bit and it is hopefully now easier to read and gives a useful error if the pipeline is not found (the dict like access already raises a key error but because we used to use get() we did not benefit from it)